### PR TITLE
Supports opening submissions in DispVMs from Qubes dev env

### PR DESCRIPTION
**Pull request by conorsch**
# Description

Adds support for running the client code (via `./run.sh`) within a Qubes AppVM, and opening submissions in DispVMs from the client interface. Related issues:

* Closes https://github.com/freedomofpress/securedrop-client/issues/482
* Closes https://github.com/freedomofpress/securedrop-client/issues/483
* Refs https://github.com/freedomofpress/securedrop-workstation/issues/288
* Refs https://github.com/freedomofpress/securedrop-debian-packaging/pull/72

# Test Plan

Throughout these instructions, `sd-dev` is assumed to be the name of the developer's AppVM where code is written. If you use a different VM name, substitute that below.

* Clone/checkout https://github.com/freedomofpress/securedrop-workstation/pull/299 and run `make all`
* In `dom0`, run `qvm-tags sd-dev add sd-client`
* In `sd-dev`, clone/checkout the "securedrop" repository and run `make dev` to start the containers
* In `sd-dev`, clone/checkout this PR branch, and run `./run.sh` to start the client interface
* Submit a PDF to the Source Interface, confirm visible in the Client
* Open the PDF submission in the Client, confirm it opens in a DispVM
* Send a reply to a source, confirm no errors

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs, network (via the RPC service) traffic, or fine tuning of the graphical user interface, Qubes testing is required. Please check as applicable:

 - [ ] I have tested these changes in Qubes
 - [ ] I do not have a Qubes OS workstation (the reviewer will need to test these changes in Qubes)

### CODE REVIEW

**Comment by creviera** [comment]
> Clone/checkout freedomofpress/securedrop-workstation#299 and run make all

freedomofpress/securedrop-workstation#299 was merged into `master` branch so this step one could clone `master` and run make all (in case there are others testing this)

**Comment by creviera** [comment]
@emkll I updated the README to include a link to the securedrop docs, mentioned the requirement to run docker as non-root user in Qubes, and an extra bit about end-to-end client testing. This is ready once again for your :eyes:

File: README.md [hunk 1]
```diff
@@ -108,14 +108,28 @@ make check
 
 ## Comparison of developer environments
 
+For end-to-end client testing on Qubes, make sure you either run the packaged client or run the unpackaged clent with proxy turned on.
```
Comment (emkll) [inline README.md]: clent -> client

File: README.md [hunk 2]
```diff
@@ -108,14 +108,28 @@ make check
...
+
 ### Developer environment on non-Qubes
 
 * Ran by `run.sh`
+* Works with SecureDrop run in a docker container, see [SecureDrop docs](https://docs.securedrop.org/en/latest/development/setup_development.html) for setup instructions
+* Uses temporary configuration directories by default
+* Requests/responses to the Journalist API are sent directly via HTTP (and Tor is not used)
```
Comment (emkll) [inline README.md]: nit: Tor also uses http, perhaps local webserver or webserver running in the local container would be clearer here?

File: README.md [hunk 3]
```diff
@@ -108,14 +108,28 @@ make check
...
+* Does not support opening submissions
+* Does not use `split-gpg`, instead uses a development gpg private key inside a gpg keychain stored in the temporary configuration directories
+
+### Developer environment on Qubes (no-proxy)
+
+* Ran by `run.sh`
+* Works with SecureDrop run in a docker container, see [SecureDrop docs](https://docs.securedrop.org/en/latest/development/setup_development.html) for setup instructions, including post-installation steps for allowing docker to be run as a non-root user, which is a requirement on Qubes
+* Requires `qvm-tags sd-dev add sd-client` to be run in `dom0` (subsitute your dev VM for `sd-dev`)
```
Comment (emkll) [inline README.md]: subsitute -> substitute

**Comment by emkll** [comment]
Opened #497 to track the use of split-gpg in dev env as a potential follow-up.

**Pull request merged**